Backpack/Chip/Classes/BPKChip.m

```diff
@@ -0,0 +1,155 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKChip.h"
+
+#import <Backpack/Font.h>
+#import <Backpack/Label.h>
+#import <Backpack/Color.h>
+#import <Backpack/Shadow.h>
+#import <Backpack/Spacing.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKChip()
+@property(nonatomic, getter=isInitializing) BOOL initializing;
+@property(nonatomic) BPKLabel* titleLabel;
+
+@property(nonatomic) UIColor *textColor;
+@end
+
+@implementation BPKChip
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    
+    if (self) {
+        [self setup];
+    }
+    
+    return self;
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    
+    if (self) {
+        [self setup];
+    }
+    
+    return self;
+}
+
+- (void)setup {
+    self.initializing = YES;
+    
+    self.titleLabel = [[BPKLabel alloc]initWithFrame:CGRectZero];
+    self.titleLabel.fontStyle = BPKFontStyleTextSm;
+    [self addSubview:self.titleLabel];
+    
+    self.titleLabel.textAlignment = NSTextAlignmentCenter;
+    
+    [self addTarget:self action:@selector(handleSingleTap:) forControlEvents:UIControlEventTouchDown];
+    
+    [self setupConstraints];
+    
+    BPKShadow *shadow = [BPKShadow shadowSm];
+    [shadow applyToLayer:self.layer];
+    
+    [self updateStyle];
+    self.initializing = NO;
+}
+
+#pragma mark - Layout overrides
+
+- (void)layoutSubviews {
+    self.layer.cornerRadius = self.bounds.size.height / 2;
+}
+
+- (CGSize)sizeThatFits:(CGSize)size {
```

Comment:
Why do we need to override this in the first place?

Reply:
Because otherwise when used in a collection view (and probably other places) I can't get the correct size for setting cell size, as it the frame is `0x0`

Reply:
Yeah - but it doesn't seem to. Maybe I'm doing something wrong 🤷‍♂️